spec/prog/postgres/postgres_server_nexus_spec.rb

```diff
@@ -897,91 +1011,194 @@
     end
 
     it "hops to wait if unfence is set" do
-      expect(nx).to receive(:when_unfence_set?).and_yield
-      expect(nx).to receive(:decr_unfence)
-      expect(postgres_server).to receive(:incr_configure)
-      expect(postgres_server).to receive(:incr_restart)
+      nx.incr_unfence
+      expect(Semaphore.where(strand_id: server_strand.id, name: "unfence").count).to eq(1)
+      expect(Semaphore.where(strand_id: server_strand.id, name: "configure").count).to eq(0)
+      expect(Semaphore.where(strand_id: server_strand.id, name: "restart").count).to eq(0)
       expect { nx.wait_in_fence }.to hop("wait")
+      # Verify unfence was decremented and configure/restart were incremented
+      expect(Semaphore.where(strand_id: server_strand.id, name: "unfence").count).to eq(0)
+      expect(Semaphore.where(strand_id: server_strand.id, name: "configure").count).to eq(1)
+      expect(Semaphore.where(strand_id: server_strand.id, name: "restart").count).to eq(1)
     end
   end
 
   describe "#prepare_for_unplanned_take_over" do
+    # Create a cluster with a real primary and standby for failover tests
+    def create_failover_cluster
+      standby_strand = described_class.assemble(
+        resource_id: postgres_resource.id, timeline_id: postgres_resource.timeline.id,
+        timeline_access: "fetch"
+      )
+      standby = PostgresServer[standby_strand.id]
+      primary = postgres_resource.servers.find { it.primary? }
+      [primary, standby, standby_strand]
```

Comment:
Probably better to just return

```suggestion
      [primary, standby]
```

and adjust callers to use `standby.strand`.